qiskit/circuit/quantumcircuit.py

```diff
@@ -566,6 +566,11 @@ def combine(self, rhs):
         for instruction_context in itertools.chain(self.data, rhs.data):
             circuit._append(*instruction_context)
         circuit.global_phase = self.global_phase + rhs.global_phase
+
+        # Add the calibrations
+        for gate, cals in rhs.calibrations.items():
+            circuit._calibrations[gate].update(cals)
```

Comment:
I think it would be more robust to call `circuit.add_calibration` here and below?

Reply:
done in b408b86.